### PR TITLE
Update c-interop-native-extensions.md

### PR DESCRIPTION
**Pull request by kwalrath**
This started out as a minor copy edit, and ended up being something of a rewrite 😆 . Since I wasn't sure about the meaning of some of the new content, I guessed. Please make sure I guessed right.

### CODE REVIEW

File: src/server/c-interop-native-extensions.md [hunk 1]
```diff
@@ -1,24 +1,23 @@
...
 
-This deprecation does not affect use-cases where the Dart VM 
-is embedded as a library into another application. 
-The Dart Embedding API (as defined in the [`include/dart_api.h`][] header) 
-remains supported for such use cases. 
-The Dart team currently does not provide any guidelines
-on how to embed the Dart VM into your application, 
-however there are [examples][] maintained by the community
-which demonstrate the necessary glue code.
+One part of the extension mechanism —
```
Comment (mraleph) [inline src/server/c-interop-native-extensions.md:15]: I am not sure I would call Embedding API "a part of the extension mechanism". The original phrasing tried to convey that "embedding" and "native extensions" are two different things even though they share some similarities and "embedding" is still supported.

Comment (kwalrath) [reply src/server/c-interop-native-extensions.md]: OK, I've tweaked the wording. ptal

**Review by mraleph — APPROVED** [review]
LGTM!

**Review by parlough — APPROVED** [review]
Thanks for cleaning this up and nice catch removing the `toc` :)

**Pull request merged**